### ISSUE TITLE
@astrojs/rss from 1.1.0 encode HTML in tags (instead of usage CDATA) and remove CDATA from customData

### ISSUE THREAD

**Issue by le0pard** [withastro/astro#5677]
### What version of `astro` are you using?

1.7.2

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Problem reproduced on @astrojs/rss >= 1.1.0.

1. @astrojs/rss till version 1.1.0 use `CDATA` to safely define any content in `description` tag. But started from 1.1.0 it just using encode all tags, if they present in XML tags. This lead, that in rss content was readable as valid HTML with `CDATA`, now it just show in some RSS readers as encoded HTML content. Added screenshots
![Screenshot 2022-12-27 at 14 39 33](https://user-images.githubusercontent.com/98444/209668122-87d406b4-0f88-43e7-bd3f-5bfb275cf6ce.png)
![Screenshot 2022-12-27 at 14 40 14](https://user-images.githubusercontent.com/98444/209668126-badd6c5b-c44e-4ae8-845a-bbfcc1ea273d.png)

2. Any tags inside `customData` with `CDATA` converted to encoded data without `CDATA` inside. Example:

```
`<itunes:subtitle><![CDATA[${_truncate(post.frontmatter.htmlAsText(), { length: 150, omission: '...' })}]]></itunes:subtitle>`,
`<itunes:summary><![CDATA[${post.frontmatter.htmlAsText()}]]></itunes:summary>`,
`<itunes:duration>${post.frontmatter.duration}</itunes:duration>`,
```

Converted to result on screenshot

![Screenshot 2022-12-27 at 14 43 12](https://user-images.githubusercontent.com/98444/209668434-279917c8-c7ba-4b84-8d23-84b8750b5cff.png)

Current solution is to use [1.0.3](https://www.npmjs.com/package/@astrojs/rss/v/1.0.3) and do not upgrade @astrojs/rss.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-jgvjth?file=src/pages/rss.xml.js

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Is this still an issue in 2.x?

**Comment by le0pard** [comment]
@matthewp yes, it is. Tested `@astrojs/rss` version 2.1.0 

No `CDATA`, just escaped all tags
<img width="1406" alt="Screenshot 2023-02-08 at 21 42 35" src="https://user-images.githubusercontent.com/98444/217634812-dd9a845a-2aa5-48f2-8488-a9727a2f59d5.png">

**Comment by ematipico** [comment]
Will look into it

**Comment by lilnasy** [comment]
This seems to be intentionally disallowed by https://github.com/withastro/astro/pull/5550.

Is there a way to add the custom data you want without undoing protections introduced by that PR?

**Comment by le0pard** [comment]
Yep, will be cool, if will be option to escape only `]]>` (need split on 2 CDATA), if it happen in content instead all html tags. But I figured myself how to fight with this (form xml myself). So I will close for now.

**Comment by JaneSmith** [comment]
This seems to still be an issue with 4.0.7. Can you clarify what your workaround was exactly?
